### PR TITLE
class modernization of library/billing.inc

### PR DESCRIPTION
**Pull request by stephenwaite**
hey @bradymiller , thought I would check in with you before continuing down path of making these functions and calls static :)

### CODE REVIEW

File: interface/billing/billing_process.php [hunk 1]
```diff
@@ -20,10 +20,10 @@
 require_once("../globals.php");
 require_once("$srcdir/patient.inc");
 require_once("$srcdir/billrep.inc");
-require_once("$srcdir/billing.inc");
 
 use OpenEMR\Billing\X12_5010_837P;
 use OpenEMR\Billing\HCFA_1500;
+use OpenEMR\Billing\Billing;
```
Comment (bradymiller) [inline interface/billing/billing_process.php]: probably would make the class name more specific (to avoid clashes in future when have more namespacing being used. Maybe something like `BillingUtilities` . Also should keep the `use` ordered alphabetically so easier in future when we have more of them.

File: interface/forms/fee_sheet/new.php [hunk 2]
```diff
@@ -11,13 +11,15 @@
...
  */
-use OpenEMR\Core\Header;
 
 require_once("../../globals.php");
 require_once("$srcdir/FeeSheetHtml.class.php");
 require_once("codes.php");
 require_once("$srcdir/options.inc.php");
 
+use OpenEMR\Core\Header;
+use OpenEMR\Billing\Billing;
```
Comment (bradymiller) [inline interface/forms/fee_sheet/new.php]: Should keep the use ordered alphabetically so easier in future when we have more of them.

File: interface/billing/billing_process.php (hunk 1 above)
Comment (stephenwaite) [reply interface/billing/billing_process.php]: ok thanks

File: interface/forms/fee_sheet/new.php (hunk 2 above)
Comment (stephenwaite) [reply interface/forms/fee_sheet/new.php]: ok, that's wise

**Comment by bradymiller** [comment]
And agree with `static` being the way to go :)

File: contrib/forms/evaluation/C_FormEvaluation.class.php [hunk 3]
```diff
@@ -64,7 +64,7 @@ function default_action_process()
```
[the remark this answers is not in the document]
Comment (stephenwaite) [reply contrib/forms/evaluation/C_FormEvaluation.class.php]: thanks

File: contrib/forms/prosthesis/C_FormProsthesis.class.php [hunk 4]
```diff
@@ -62,7 +62,7 @@ function default_action_process()
```
[the remark this answers is not in the document]
Comment (stephenwaite) [reply contrib/forms/prosthesis/C_FormProsthesis.class.php]: 👌

File: contrib/forms/review_of_systems/C_FormReviewOfSystems.class.php [hunk 5]
```diff
@@ -62,7 +62,7 @@ function default_action_process()
```
[the remark this answers is not in the document]
Comment (stephenwaite) [reply contrib/forms/review_of_systems/C_FormReviewOfSystems.class.php]: yup, think php storm was trying to tell me this when I went to autocomplete the :: use and it gave me a full path 🤷‍♂️

File: library/billing.inc [hunk 6]
```diff
@@ -1,539 +0,0 @@
-<?php
-
```
[the remark this answers is not in the document]
Comment (stephenwaite) [reply library/billing.inc]: 👋

File: library/billing/src/HCFA_1500.php [hunk 7]
```diff
@@ -541,7 +542,7 @@ private function gen_hcfa_1500_page($pid, $encounter, &$log, $claim)
                     "' is not justified!\n";
             }
 
-            $clm_total_charges += floatval($claim->cptCharges($this->hcfa_proc_index));
+            $clm_total_charges += floatval()$claim->cptCharges($this->hcfa_proc_index);
```
[the remark this answers is not in the document]
Comment (stephenwaite) [reply library/billing/src/HCFA_1500.php]: nope, was trying to cast this like the use of int() seen elsewhere so reverted

**Comment by bradymiller** [comment]
hi @stephenwaite , Just let me know if you think this is ready to come into codebase. I'll fix the merge conflicts before I bring it in. thanks, -brady

**Comment by stephenwaite** [comment]
thank you @bradymiller , it's ready to go

**Comment by stephenwaite** [comment]
couldn't resist another fix :)

File: interface/main/authorizations/authorizations.php [hunk 8]
```diff
@@ -12,13 +12,13 @@
 
 require_once("../../globals.php");
```
Comment (bradymiller) [inline interface/main/authorizations/authorizations.php]: TODO
Check if this script uses any billing functions?

File: interface/patient_file/report/custom_report.php [hunk 9]
```diff
@@ -15,7 +15,6 @@
 
 require_once("../../globals.php");
```
Comment (bradymiller) [inline interface/patient_file/report/custom_report.php]: TODO
Check if this script uses any billing functions

File: interface/reports/edi_270.php [hunk 10]
```diff
@@ -18,12 +18,12 @@
 
 require_once("../globals.php");
```
Comment (bradymiller) [inline interface/reports/edi_270.php]: TODO
Check if this script uses any billing functions

File: interface/reports/edi_271.php [hunk 11]
```diff
@@ -13,12 +13,12 @@
 
 require_once(dirname(__file__)."/../globals.php");
 require_once("$srcdir/forms.inc");
```
Comment (bradymiller) [inline interface/reports/edi_271.php]: TODO
Check if this script uses any billing functions

File: interface/patient_file/printed_fee_sheet.php [hunk 12]
```diff
@@ -16,8 +16,9 @@
 require_once("../globals.php");
 require_once("$srcdir/acl.inc");
```
Comment (bradymiller) [inline interface/patient_file/printed_fee_sheet.php]: TODO
Check if this script uses any billing functions

File: myportal/soap_service/server_med_rec.php [hunk 13]
```diff
@@ -314,7 +315,7 @@ private function getIncudes($val)
         $copays = 0.00;
         foreach ($ar['newpatient'] as $be) {
             $ta = split(":", $be);
-            $billing = getPatientBillingEncounter($pid, $ta[1]);
+            $billing = BillingUtilities::getBillingByEncounter($pid, $ta[1]);
```
Comment (bradymiller) [inline myportal/soap_service/server_med_rec.php]: TODO
revert the changes in this script since the function getPatientBillingEncounter() is in report.inc and also check if any billing functions are being called in this script

File: portal/report/portal_custom_report.php [hunk 14]
```diff
@@ -43,7 +43,6 @@
 
 require_once('../../interface/globals.php');
 require_once("$srcdir/forms.inc");
```
Comment (bradymiller) [inline portal/report/portal_custom_report.php]: TODO
Check to see if any billing functions are being called in this script

**Comment by bradymiller** [comment]
@stephenwaite ,
Nice work! Gonna bring this in and then make a couple minor fixes. Keep those classes coming :)
-brady

**Comment by stephenwaite** [comment]
thanks again for the help and then tidying up for me :)
